build: Update libmultiprocess library

hebasto opened the pull request:
This update in particular includes:
- https://github.com/chaincodelabs/libmultiprocess/pull/78 which is [required](https://github.com/bitcoin/bitcoin/pull/25972#issuecomment-1246812573) for https://github.com/bitcoin/bitcoin/pull/25972
- https://github.com/chaincodelabs/libmultiprocess/pull/74
- https://github.com/chaincodelabs/libmultiprocess/pull/70

hebasto committed:
build: Update `libmultiprocess` library

Replacing `install` with `install-lib` and `install-bin` is not strictly
necessary just to update the library, but it takes advantage of recent
changes in the new version, and makes the build more minimal.

ryanofsky commented on depends/packages/libmultiprocess.mk:
```diff
@@ -24,5 +24,5 @@ define $(package)_build_cmds
 
 define $(package)_stage_cmds
-  $(MAKE) DESTDIR=$($(package)_staging_dir) install
+  $(MAKE) DESTDIR=$($(package)_staging_dir) install-lib
```
In commit "build: Update libmultiprocess library" (2a2dd0a80c61792ad30a7814dff55a9d46391d91)

Note for reviewers: replacing `install` with `install-lib` and `install-bin` here isn't strictly necessary just to update the library, but it takes advantage of recent changes in the new version, and makes the build more minimal.

hebasto replied:
> This could be noted in the commit message, too.

Thanks! The commit message has been updated.

ryanofsky reviewed (approved):
Code review ACK 2a2dd0a80c61792ad30a7814dff55a9d46391d91.

Thanks for the version update. I do want to look into converting the library into a subtree so this step isn't necessary in the future.

hebasto commented:
Updated 2a2dd0a80c61792ad30a7814dff55a9d46391d91 -> 1986f129c6af7644d238c76f032713731d7afcac ([pr26672.01](https://github.com/hebasto/bitcoin/commits/pr26672.01) -> [pr26672.02](https://github.com/hebasto/bitcoin/commits/pr26672.02), [diff](https://github.com/hebasto/bitcoin/compare/pr26672.01..pr26672.02)):

- addressed @ryanofsky's [comment](https://github.com/bitcoin/bitcoin/pull/26672#discussion_r1044558944)

ryanofsky reviewed (approved):
Code review ACK 1986f129c6af7644d238c76f032713731d7afcac

Just updated commit message since last review (thanks!)
